homeassistant/components/solarlog/sensor.py

```diff
@@ -39,4 +39,9 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the native sensor value."""
-        return self.coordinator.data[self.entity_description.json_key]
+        return (
+            getattr(self.coordinator.data, self.entity_description.key)
+            * self.entity_description.factor
+            if self.entity_description.factor
+            else getattr(self.coordinator.data, self.entity_description.key)
```

Comment:
We try to avoid ternary operator expression if it spans more than one line.

Reply:
Ok, I added a rounding option to limit the number of digits of the percentage measurements (it had about 10 digits). Does not affect the other measurements with `round(x, 3)`